Components/Inputs/TLV2544.cs

```diff
@@ -0,0 +1,264 @@
+﻿using System;
+using System.Threading;
+using Scarlet.IO;
+
+namespace Scarlet.Components.Inputs
+{
+    /// <summary>
+    /// 200kSa/s 4-ch Analogue to Digital Converter
+    /// Datasheet: http://www.ti.com/lit/ds/symlink/tlv2544.pdf
+    /// </summary>
+    public class TLV2544
+    {
+        public class AnalogueInTLV254x : IAnalogueIn
+        {
+            private TLV2544 Parent;
+            private byte Channel;
+
+            public AnalogueInTLV254x(TLV2544 Parent, byte Channel)
+            {
+                this.Parent = Parent;
+                this.Channel = Channel;
+            }
+
+            /// <summary> Gets the current ADC input level. </summary>
+            /// <returns> The current input in Volts. </returns>
+            public double GetInput() => ((double)(this.Parent.GetRawInput(this.Channel)) / GetRawRange()) * GetRange();
+
+            /// <summary> Gets the maximum input value. Depends on voltage reference used. </summary>
+            /// <returns> The maximum input voltage, in Volts. </returns>
+            public double GetRange() => this.Parent.GetRange();
+
+            /// <summary> Gets the raw digital input value, without voltage reference scaling. </summary>
+            /// <returns> The raw ADC input, in Volt-bits. Between 0 and <c>GetRawRange()</c>. </returns>
+            public long GetRawInput() => this.Parent.GetRawInput(this.Channel);
+
+            /// <summary> The maximum digital value that the ADC can output. THis is fixed per chip type. </summary>
+            /// <returns> The maximum an unsigned 12bit number can hold, 4095, as this is a 12b ADC. </returns>
+            public long GetRawRange() => 4095;
+
+            public void Dispose() { }
+        }
+
+        public static readonly Configuration DefaultConfig = new Configuration()
+        {
+            VoltageRef = VoltageReference.INTERNAL_4V,
+            UseLongSample = false,
+            ConversionClockSrc = ConversionClockSrc.SCLK,
+            ConversionMode = ConversionMode.SINGLE_SHOT,
+            UseEOCPin = false,
+            FIFOTriggerLevel = FIFOTrigger.FIFO_8b
+        };
+        public readonly AnalogueInTLV254x[] Inputs;
+
+        private readonly ISPIBus Bus;
+        private readonly IDigitalOut CS;
+        private Configuration Config = DefaultConfig;
+        private sbyte ReuseChannel = -1; // TODO: Implement channel re-use to speed up repeated channel reads.
```

Comment:
Are we going to wait on this feature?

Reply:
This is a holdover from when I was planning on implementing auto-sampling, it has no utility when using single-shot, so I've removed it.